app/uk/gov/hmrc/agentsubscriptionfrontend/controllers/SubscriptionController.scala

```diff
@@ -112,43 +110,43 @@ class SubscriptionController @Inject()
+                      address: DesAddress, addressLookupAddress: AddressLookupFrontendAddress): Future[Either[SubscriptionReturnedHttpError, (Arn, String)]] = {
           val subscriptionDetails = mapper(details, address)
           subscriptionService.subscribeAgencyToMtd(subscriptionDetails) map {
             case Right(arn) => {
-              if (addressLookupAddress.lines.length > 4) Logger.warn("UTR with more than 4 address lines: " + details.utr.value)
               Right((arn, subscriptionDetails.name))
             }
             case Left(x) => Left(SubscriptionReturnedHttpError(x))
           }
         }
 
-        addressLookUpConnector.getAddressDetails(id).flatMap { address =>
-          validateAddress(address, blacklistedPostCodes) match {
-            case Invalid(errors) =>
-              Future.successful(
-                Ok(uk.gov.hmrc.agentsubscriptionfrontend.views.html.des_will_not_accept_address(id, SubscriptionController.renderErrors(errors)))
-              )
-            case Valid(desAddress) =>
-              val subscriptionResponse = for {
-                detailsOpt <- sessionStoreService.fetchInitialDetails
-                subscriptionResponse <- detailsOpt match {
-                  case Some(details) => subscribe(details, desAddress, address)
-                  case None => Future.successful(Left(MissingSessionData))
-                }
-                _ <- sessionStoreService.remove()
-              } yield subscriptionResponse
-
-              subscriptionResponse.map {
-                case Right((arn, agencyName)) => Redirect(routes.SubscriptionController.showSubscriptionComplete())
-                  .flashing("arn" -> arn.arn, "agencyName" -> agencyName)
-                case Left(SubscriptionReturnedHttpError(CONFLICT)) => Redirect(routes.CheckAgencyController.showAlreadySubscribed())
-                case Left(SubscriptionReturnedHttpError(FORBIDDEN)) => Redirect(routes.SubscriptionController.showSubscriptionFailed())
-                case Left(MissingSessionData) => sessionMissingRedirect()
-                case Left(error) => InternalServerError(s"Unknown error code from agent-subscription $error")
-              }
+        def redirectSubscriptionResponse(either: Either[SubscriptionReturnedHttpError, (Arn, String)]): Result = {
+          either match {
+            case Right((arn, agencyName)) => Redirect(routes.SubscriptionController.showSubscriptionComplete())
+              .flashing("arn" -> arn.arn, "agencyName" -> agencyName)
+            case Left(SubscriptionReturnedHttpError(CONFLICT)) => Redirect(routes.CheckAgencyController.showAlreadySubscribed())
+            case Left(SubscriptionReturnedHttpError(_)) => Redirect(routes.SubscriptionController.showSubscriptionFailed())
           }
         }
+
+        sessionStoreService.fetchInitialDetails.flatMap { maybeDetails =>
```

Comment:
Something seems to have gone wrong with this commit - these changes (and some others) are already in master.